backend/resources/books.py

```diff
@@ -171,4 +171,12 @@ def get(self):
                             help="In this field goes the number of the books to show")
         data = parser.parse_args()
         isbns = TransactionsModel.best_sellers()
-        return {'books': [BooksModel.find_by_isbn(book).json(score=True) for book in isbns[:data['numBooks']]]}, 200
+        books = []
+        for isbn in isbns:
+            if len(books) == data['numBooks']:
+                return {'books': books}, 200
+            book = BooksModel.find_by_isbn(isbn)
+            if book.vendible:
+                books.append(BooksModel.find_by_isbn(isbn).json(score=True))
+
+        return {'books': books}, 200
```

Comment:
Esta be pero el if del loop queda extrany, seria millor que possesis un `break` perque tens la mateixa sortida que el cas normal. Et proposo una possible solucio mes compacte.
```suggestion
        return {'books': filter(lambda book: book['venidble'], map(lambda x: BooksModel.find_by_isbn(x).json(score=True), isbns))[:data['numBooks']]]}, 200
```